src/core/service.c

```diff
@@ -1358,10 +1372,69 @@ static int service_collect_fds(
                 rfd_names[n_fds] = NULL;
         }
 
+        if (s->open_files) {
+                size_t n_fds = rn_socket_fds + s->n_fd_store;
+
+                LIST_FOREACH(open_files, f, s->open_files) {
+                        struct stat st;
+                        _cleanup_close_ int fd = -1;
+
+                        if (stat(f->path, &st) < 0)
+                                return log_error_errno(errno, "Failed to stat %s: %m", f->path);
+
+                        if (S_ISSOCK(st.st_mode)) {
+                                struct sockaddr_un addr;
+                                socklen_t sa_len;
+
+                                fd = socket(AF_UNIX, SOCK_STREAM|SOCK_CLOEXEC, 0);
+                                if (fd < 0) {
+                                        log_unit_error_errno(UNIT(s), errno, "Failed to create socket for %s: %m", f->path);
+                                        continue;
+                                }
+
+                                r = sockaddr_un_set_path(&addr, f->path);
+                                if (r < 0) {
+                                        log_unit_error_errno(UNIT(s), errno, "Failed to fill sockaddr for %s: %m", f->path);
+                                        continue;
+                                }
+
+                                sa_len = r;
+                                r = connect(fd, (struct sockaddr*)&addr, sa_len);
```

Comment:
we might as well try all three of SOCK_DGRAM, SOCK_STREAM and SOCK_SEQPACKET, one after the other.